src/nvim/channel.c

```diff
@@ -46,7 +46,8 @@ static bool did_stdio = false;
 /// next free id for a job or rpc channel
 /// 1 is reserved for stdio channel
 /// 2 is reserved for stderr channel
-static uint64_t next_chan_id = CHAN_STDERR + 1;
+/// 3 is expected to be first socket listener connection
```

Comment:
Ok, i can see that - there's definitely a race on first channel. Theres an issue in that the numbers of the listen channels aren't allocated until connect. So I'm not quite sure how to loop on a wait for a ui connect - other than that I could avoid checking for specific channels and instead check if any ui connected. However the event streams are channel specific. I assume it would be possible to loop all channels in a listen block - maybe that the right approach rather than serially polling specific ones?

Reply:
in that case you would watch the main event queue `main_loop.events`. This will implicitly include all rpc channels as child queues.

Reply:
great! Made that mod.  Thanks